src/components/REPLInput.js

```diff
@@ -32,12 +33,15 @@ export class REPLInput extends React.Component {
         <span style={styles.inputChevron}>
           {'> '}
         </span>
-        <input
-          style={styles.input}
+        <textarea
+          style={styles.textarea}
           value={this.props.code}
           onChange={event => this.props.changeREPLCode(event.target.value)}
-          onKeyPress={event =>
-                 event.key === "Enter" && this.props.onRun(this.props.code)}
+          onKeyPress={event => {
+            (event.shiftKey && event.key === "Enter") ?
+            this.props.changeREPLCode(event.target.value) :
+            event.key === "Enter" && this.props.onRun(this.props.code);
+          }}
```

Comment:
I would probably just put this whole onKeyPress into a standalone class method. Could be something like:

```jsx
handleKeyPress = (event) => {
  ...
}
```
and then in the jsx:
```jsx
onKeyPress={this.handleKeyPress}
```